test/org/sagebionetworks/bridge/services/SurveyServiceMockTest.java

```diff
@@ -120,7 +129,35 @@ public void successfulDeletePermanently() {
         verify(mockSurveyDao).deleteSurveyPermanently(keysCaptor.capture());
         assertEquals(survey, keysCaptor.getValue());
     }
-    
+
+    @Test(expected = BadRequestException.class)
+    public void logicallyDeleteSurveyNotEmptySharedModules() {
+        when(mockSharedModuleMetadataService.queryAllMetadata(anyBoolean(), anyBoolean(), anyString(), anySetOf(String.class)))
+                .thenReturn(ImmutableList.of(makeValidMetadata()));
+
+        Survey survey = createSurvey();
+        doReturn(survey).when(mockSurveyDao).getSurvey(any());
+        service.deleteSurvey(survey);
+    }
+
+    @Test(expected = BadRequestException.class)
+    public void physicallyDeleteSurveyNotEmptySharedModules() {
+        when(mockSharedModuleMetadataService.queryAllMetadata(anyBoolean(), anyBoolean(), anyString(), anySetOf(String.class)))
+                .thenReturn(ImmutableList.of(makeValidMetadata()));
+
+        List<SchedulePlan> plans = createSchedulePlanListWithSurveyReference(false);
+
+        Activity oldActivity = getActivityList(plans).get(0);
+        Activity activity = new Activity.Builder().withActivity(oldActivity)
+                .withSurvey("Survey", "otherGuid", SURVEY_CREATED_ON).build();
+        getActivityList(plans).set(0, activity);
+
+        doReturn(plans).when(mockSchedulePlanService).getSchedulePlans(ClientInfo.UNKNOWN_CLIENT, TEST_STUDY);
```

Comment:
It would be better to just return an empty list of plans, since we don't need to exercise that logic here.